src/server/game/Achievements/AchievementMgr.cpp

```diff
@@ -2289,6 +2284,18 @@ void AchievementGlobalMgr::LoadAchievementCriteriaList()
             }
         }
 
+        for (uint32 conditionField = 0; conditionField < MAX_CRITERIA_REQUIREMENTS; ++conditionField)
+        {
+            if (criteria->AdditionalRequirements[conditionField].Type != ACHIEVEMENT_CRITERIA_CONDITION_NONE)
+            {
+                ASSERT(criteria->AdditionalRequirements[conditionField].Type < ACHIEVEMENT_CRITERIA_CONDITION_MAX, "ACHIEVEMENT_CRITERIA_CONDITION_MAX must be greater than or equal to %u but is currently equal to %u",
+                    criteria->AdditionalRequirements[conditionField].Type + 1, ACHIEVEMENT_CRITERIA_CONDITION_MAX);
+
+                if (conditionField == 0 || criteria->AdditionalRequirements[conditionField].Type != criteria->AdditionalRequirements[conditionField-1].Type)
```

Comment:
No, but I may change the if to check for that (patchs > 3.3.5 may differ in this matter)